Lang tag for checklists/rpc/linkvoucher

NikitaSalikov opened the pull request:
# Changes Made:

- Internationalized text

# Pre-Approval

- [X] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [N/A] Hotfixes should be branched off of the `master` branch and **squash and merged** back into the `master` branch.
- [X] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [X] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [N/A] There are no linter errors
- [X] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [X] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [N/A] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [X] Comment which GitHub issue(s), if any does this PR address:
https://github.com/orgs/BioKIC/projects/6/views/14?pane=issue&itemId=51314274
- [N/A] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [x] It is the code author's responsibility to merge their own pull request after it has been approved
- [x] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a hotfix into the `master` branch, a subsequent PR from `master` into `Development` should be made **merge** option (i.e., no squash).
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [x] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

NikitaSalikov committed:
internationalized text

NikitaSalikov committed:
added es and fr text

NikitaSalikov committed:
fixed slash typo

themerekat reviewed (changes requested):
@NikitaSalikov , shouldn't the language files be named linkvoucher.en.php , etc.?

Atticus29 commented:
> If you want, I suggest improving some bad code practice that was previously applied here. I think it would be better if the rpc file only returned a json object containing a status code and the error message upon fail. This would allow the language translations to be included within the ("view") file making use of the rpc handler. If there are multiple pages using this AJAX handler, they might have different ways of dealing with success and errors.
> 
> Upon success, maybe something like: $retJson = '{"status": 1}'; And upon failure: $retJson = '{"status": 0, "error": "' . $clManager->getErrorMessage() . '"}';
> 
> You will of course need to modify the source file making us of this handler. If you have a better idea, please feel free to apply. Note that if the source file is a .js file, it might be difficult adding the LANG messages. If that is the case, the js can be moved into php file calling this handler. Or we can consider another option.

@egbot per Feb. 5, 2024 Dev meeting, would you be ok implementing an example of this so that Nikita and I know what you're looking for in future PRs?

egbot committed:
Misc adjustments

After future investigation, I learned that this RPC is used by multiple calls within checklist and collection tools with the same messaging used within each. Thus, I agree that it is probably best to keep the translation within the RPC file, rather than transferring a code that is translated. I did move the translation file into the file structure suggested by Katie, to match current protocols. I also did additional cleaning, adjustments, etc beyond just translating.
- Add default when input variables are null
- Add translation for when voucher is already link to taxon
- Added minor adjustment to another translation file. Google translate incorrectly translates voucher to cupón (coupon). Not sure if there is a better translation. Ideally, all Google translation would be verified by an expert.
